pkg/stream/consumer.go

```diff
@@ -14,68 +14,105 @@ import (
 
 const metricNameConsumerProcessedCount = "ConsumerProcessedCount"
 
+//go:generate mockery -name=Consumer
+type Consumer interface {
+	Boot(config cfg.Config, logger mon.Logger) error
+	Run(ctx context.Context) error
+	GetType() string
+}
+
 //go:generate mockery -name=ConsumerCallback
 type ConsumerCallback interface {
 	Boot(config cfg.Config, logger mon.Logger) error
 	Consume(ctx context.Context, msg *Message) (bool, error)
 }
 
-type Consumer struct {
+type ConsumerSettings struct {
+	Name        string
+	IdleTimeout time.Duration
+	RunnerCount int
+}
+
+type consumer struct {
 	kernel.EssentialModule
 	ConsumerAcknowledge
 
-	logger mon.Logger
-	mw     mon.MetricWriter
+	metric mon.MetricWriter
 	tracer tracing.Tracer
 	cfn    coffin.Coffin
 	ticker *time.Ticker
 
-	name      string
-	callback  ConsumerCallback
+	settings ConsumerSettings
+	callback ConsumerCallback
+
 	processed int32
 }
 
-func NewConsumer(callback ConsumerCallback) *Consumer {
-	return &Consumer{
+func NewConsumer(callback ConsumerCallback) Consumer {
+	return &consumer{
 		cfn:      coffin.New(),
 		callback: callback,
 	}
 }
 
-func (c *Consumer) Boot(config cfg.Config, logger mon.Logger) error {
+func NewConsumerWithInterfaces(callback ConsumerCallback, logger mon.Logger, tracer tracing.Tracer, input Input, metricWriter mon.MetricWriter, settings ConsumerSettings) Consumer {
+	consumer := NewConsumer(callback).(*consumer)
+
+	consumer.bootWithInterfaces(logger, tracer, input, metricWriter, settings)
+
+	return consumer
+}
+
+func (c *consumer) Boot(config cfg.Config, logger mon.Logger) error {
 	err := c.callback.Boot(config, logger)
 
 	if err != nil {
 		return err
 	}
 
-	appId := cfg.GetAppIdFromConfig(config)
-	c.name = fmt.Sprintf("consumer-%v-%v", appId.Family, appId.Application)
-
-	c.logger = logger
-	c.tracer = tracing.NewAwsTracer(config)
+	tracer := tracing.NewAwsTracer(config)
 
 	defaultMetrics := getConsumerDefaultMetrics()
-	c.mw = mon.NewMetricDaemonWriter(defaultMetrics...)
-
-	idleTimeout := config.GetDuration("consumer_idle_timeout")
-	c.ticker = time.NewTicker(idleTimeout * time.Second)
+	mw := mon.NewMetricDaemonWriter(defaultMetrics...)
 
 	inputName := config.GetString("consumer_input")
 	input := NewConfigurableInput(config, logger, inputName)
 
-	c.input = input
-	c.ConsumerAcknowledge = NewConsumerAcknowledgeWithInterfaces(logger, input)
+	idleTimeout := config.GetDuration("consumer_idle_timeout") * time.Second
```

Comment:
here we use consumer_idle_timeout multiplied with time.Second, in the other occurence we don't. Are we having a bug here?